thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1230,15 +1214,15 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
                     // If !preserverOrder, right rows added to single array in jg, so pass 0
                     joinGroup->addRightMatchCompletePending(activity.preserveOrder ? indexPartNo : 0, sequence, fetchRow);
 
-                    activity.statsArr[AS_DiskAccepted]++;
-                    if (activity.statsArr[AS_DiskAccepted] > activity.rowLimit)
+                    diskAccepted++;
+                    if (diskAccepted > activity.rowLimit)
```

Comment:
I think this needs to be diskAccepted + activity.statsArr[AS_DiskAccepted], but I think what you are expecting is more natural.  @richardkchapman what do you think about changing it?